src/app/visitor/account/tokens.controller.js

```diff
@@ -0,0 +1,138 @@
+angular.module("visitorModule")
+
+.controller('visitorAccountTokensController', [
+    '$scope',
+    '$location',
+    '_',
+    'visitorLoginService',
+    'visitorApiService',
+    'checkoutService',
+    'commonUtilService',
+    function(
+        $scope,
+        $location,
+        _,
+        visitorLoginService,
+        visitorApiService,
+        checkoutService,
+        commonUtilService
+    ) {
+        // General
+        $scope.visitor = visitorLoginService.getVisitor();
+
+        // Tokens List
+        $scope.tokens = [];
+        $scope.remove = remove;
+        $scope.getTokens = getTokens;
+        $scope.popUpOpen = popUpOpen;
+        $scope.showTokenMethod = false;
+        $scope.tokenAbleMethods = [];
+        $scope.getPaymentMethodName = getPaymentMethodName;
+
+        // Edit / Add Form
+        $scope.token = {};
+        $scope.message = '';
+        $scope.save = save;
+        $scope.parseDate = parseDate;
+        $scope.creditCardData = {};
+
+        // Sidebar, refactor
+        var activePath = $location.path();
+        $scope.isActive = isActive;
+
+        activate();
+
+        //////////////////////////
+
+        function activate() {
+            // BREADCRUMBS
+            $scope.$emit('add-breadcrumbs', {
+                'label': 'myAccount',
+                'url': '/account'
+            });
+            $scope.$emit('add-breadcrumbs', {
+                'label': 'Credit Cards',
+                'url': '/account/creditcards'
+            });
+
+            // Redirect
+            visitorLoginService.isLoggedIn()
+                .then(function(isLoggedIn) {
+                    if (!isLoggedIn) {
+                        $location.path("/");
+                    }
+                });
+
+            $scope.getTokens();
+
+            checkoutService.loadPaymentMethods().then(function(methods){
+                if (methods.length == 0) {
```

Comment:
we can use lodash to filter payment methods by `tokenable` field, like `$scope.tokenAbleMethods = _.filter(methods, { 'Tokenable': true });`